components/Flashcard/Flashcard.js

```diff
@@ -1,30 +1,100 @@
 import styled from "styled-components";
+import { useState } from "react";
 
 export default function Flashcard({ question, answer, collection }) {
+  const [showAnswer, setShowAnswer] = useState(false);
+
+  function handleShowAnswer() {
+    setShowAnswer(!showAnswer);
+  }
+
   return (
-    <StyledFlashcard>
-      <CollectionTitle>{collection}</CollectionTitle>
-      <Question>{question}</Question>
-      <p>{answer}</p>
-    </StyledFlashcard>
+    <CardContainer onClick={handleShowAnswer}>
+      <StyledFlashcard $showAnswer={showAnswer}>
+        <CardFront>
+          <CollectionTitle>{collection}</CollectionTitle>
+          <Question>{question}</Question>
+          <StyledArrow
+            xmlns="http://www.w3.org/2000/svg"
+            height="24px"
+            viewBox="0 -960 960 960"
+            width="24px"
+            fill="#000"
+          >
+            <path d="m700-300-57-56 84-84H120v-80h607l-83-84 57-56 179 180-180 180Z" />
+          </StyledArrow>
+        </CardFront>
+        <CardBack>
+          <Answer>{answer}</Answer>
+          <StyledArrow
+            xmlns="http://www.w3.org/2000/svg"
+            height="24px"
+            viewBox="0 -960 960 960"
+            width="24px"
+            fill="#000"
+            transform="scale(-1 1)"
+          >
+            <path d="m700-300-57-56 84-84H120v-80h607l-83-84 57-56 179 180-180 180Z" />
+          </StyledArrow>
+        </CardBack>
+      </StyledFlashcard>
+    </CardContainer>
   );
 }
 
+const CardContainer = styled.div`
+  perspective: 1000px;
+`;
+
 const StyledFlashcard = styled.section`
+  width: 19rem;
+  height: 13rem;
+  position: relative;
+  transform-style: preserve-3d;
+  transition: transform 0.6s;
+  transform: ${({ $showAnswer }) =>
+    $showAnswer ? "rotateY(180deg)" : "rotateY(0deg)"};
+  cursor: pointer;
+`;
+
+const CardFace = styled.div`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  backface-visibility: hidden;
   border: 1px solid #000;
   padding: 10px;
+  border-radius: 10px;
+`;
+
+const CardFront = styled(CardFace)`
+  background-color: #fff;
+`;
+
+const CardBack = styled(CardFace)`
+  background-color: #e6e6e6;
+  transform: rotateY(180deg);
 `;
 
 const CollectionTitle = styled.p`
   font-size: 0.8rem;
 `;
 
-const Question = styled.p`
+const CardText = styled.p`
   font-size: 1.2rem;
   font-weight: 500;
 `;
 
-const Answer = styled.p`
-  font-size: 1rem;
-  font-weight: 500;
+const Question = styled(CardText)`
```

Comment:
I feel like the question should be a headline instead of a `p` element. 🤔

Reply:
We have adapted the code accordingly.